relay-metrics/src/protocol.rs

```diff
@@ -164,20 +165,92 @@ fn is_valid_name(name: &str) -> bool {
     false
 }
 
-/// Validates an MRI of the form `<ty>:<ns>/<name>@<unit>`
+/// The namespace of a metric.
 ///
-/// Note that the format used in the statsd protocol is different: Metric names are not prefixed
-/// with `<ty>:` as the type is somewhere else in the protocol.
-pub(crate) fn is_valid_mri(name: &str) -> bool {
-    let mut components = name.splitn(2, ':');
-    if components.next().is_none() {
-        return false;
+/// This successfully deserializes any kind of string, but in reality only `"sessions"` and
+/// `"transactions"` is supported. Everything else is dropped both in the metrics aggregator and in
+/// the store actor.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum MetricNamespace {
+    /// Metrics extracted from sessions.
+    Sessions,
+    /// Metrics extracted from transaction events.
+    Transactions,
+    /// Metrics that relay doesn't know the namespace of, and will drop before aggregating.
```

Comment:
_nit_: This can be misinterpreted. Primarily, this variant is used for unsupported namespaces. For instance if you send an MRI `c:something_new/foo@none`. Relay knows the namespace is "something_new", it just doesn't support it.